test/unit/org/apache/cassandra/metrics/TrieMemtableMetricsTest.java

```diff
@@ -130,14 +130,26 @@ public void testFlushRelatedMetrics() throws IOException, ExecutionException, In
         assertEquals(0, metrics.contendedPuts.getCount() + metrics.uncontendedPuts.getCount());
 
         writeAndFlush(10);
+        cfs.waitForReclamations();  // wait for lastFlushShardDataSizes update to be applied and visible
```

Comment:
I'm wary about making dependency between the metrics and reclamation so explicit, and adding a specific test-only method to CFS...

I don't insist, but WDYT about using code like
```        await().atMost(5, TimeUnit.SECONDS)
               .until(() -> previousMaxShardDataSizes < metrics.lastFlushShardDataSizes.maxGauge.getValue());
```
instead?

Reply:
This does sound like a good enough solution for this (even though it still runs some low chance of timeout), changed.

Reply:
Out of curiosity Is the 30s `timeout` really possible? I mean - `writeAndFlush` already waits until the flush ends, so it's only reclaim (wait for barrier, memtable.discard()) that we'll be waiting for.

Reply:
No. Even one would normally be overdoing it.

Unfortunately I've seen time and again things that should asynchronously happen in microseconds take 10s of seconds on a CI test run. I guess due to running many tests on the same machine and one runner being on pause for much longer than expected.